include/seqan3/core/platform.hpp

```diff
@@ -34,20 +44,29 @@
 //  C++ standard and features
 // ============================================================================
 
+#if SEQAN3_DOXYGEN_ONLY(1)0
+//!\brief This disables the warning you would get if you compile with `-std=c++17`.
+#define SEQAN3_DISABLE_CPP17_DIAGNOSTIC
```

Comment:
I would maybe write something a little more directed at the user, like "define this variable to disable..."

Side note: We do not really define this variable, but we need to define it so we can document it?